src/messages/MOSDOpReply.h

```diff
@@ -194,7 +194,33 @@ class MOSDOpReply : public Message {
   }
   virtual void decode_payload() {
     bufferlist::iterator p = payload.begin();
-    if (header.version < 2) {
+
+    // Always keep here the newest version of decoding order/rule
+    if (header.version == HEAD_VERSION) {
+		::decode(oid, p);
+		::decode(pgid, p);
+		::decode(flags, p);
+		::decode(result, p);
+		::decode(bad_replay_version, p);
+		::decode(osdmap_epoch, p);
+
+		__u32 num_ops = ops.size();
+		::decode(num_ops, p);
+		ops.resize(num_ops);
+		for (unsigned i = 0; i < num_ops; i++)
+		::decode(ops[i].op, p);
+		::decode(retry_attempt, p);
+
+		for (unsigned i = 0; i < num_ops; ++i)
+		::decode(ops[i].rval, p);
+
+		OSDOp::split_osd_op_vector_out_data(ops, data);
+
+		::decode(replay_version, p);
+		::decode(user_version, p);
+		::decode(redirect, p);
+
```

Comment:
looks like the indentation is off... emacs and vi should use the modeline at the top of the file.